File: app/src/main/java/com/android/example/popularmovies/utils/network/OkHttpException.java
package com.android.example.popularmovies.utils.network;

import androidx.annotation.Nullable;

import okhttp3.Response;

public class OkHttpException extends RuntimeException {
    private Response response;

    @Nullable
    @Override
    public String getMessage() {
        switch (response.code()){
            case 400:
                return "Invalid request syntax";
            case 401:
                return "Not authorized to access this information";
            case 404:
                return "The server could not be found";
            default:
                return super.getMessage();
        }

    }

    public OkHttpException(Response response) {
        this.response = response;
    }
}

File: app/src/main/java/com/android/example/popularmovies/utils/network/OkHttpException.kt
package com.android.example.popularmovies.utils.network

import okhttp3.Response

class OkHttpException(private val response: Response) : RuntimeException() {
    override val message: String?
        get() = when (response.code) {
            400 -> "Invalid request syntax"
            401 -> "Not authorized to access this information"
            404 -> "The server could not be found"
            else -> super.message
        }

}